plugins/catalog-node/report.api.md

```diff
@@ -102,6 +119,94 @@ export type CatalogProcessorResult =
   | CatalogProcessorErrorResult
   | CatalogProcessorRefreshKeysResult;
 
+// @public
+export interface CatalogService extends CatalogApi {
+  // (undocumented)
+  addLocation(
+    location: AddLocationRequest,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<AddLocationResponse>;
+  // (undocumented)
+  getEntities(
+    request?: GetEntitiesRequest,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<GetEntitiesResponse>;
+  // (undocumented)
+  getEntitiesByRefs(
+    request: GetEntitiesByRefsRequest,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<GetEntitiesByRefsResponse>;
+  // (undocumented)
+  getEntityAncestors(
+    request: GetEntityAncestorsRequest,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<GetEntityAncestorsResponse>;
+  // (undocumented)
+  getEntityByRef(
+    entityRef: string | CompoundEntityRef,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<Entity | undefined>;
+  // (undocumented)
+  getEntityFacets(
+    request: GetEntityFacetsRequest,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<GetEntityFacetsResponse>;
+  // (undocumented)
+  getLocationByEntity(
+    entityRef: string | CompoundEntityRef,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<Location_2 | undefined>;
+  // (undocumented)
+  getLocationById(
+    id: string,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<Location_2 | undefined>;
+  // (undocumented)
+  getLocationByRef(
+    locationRef: string,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<Location_2 | undefined>;
+  // (undocumented)
+  queryEntities(
+    request?: QueryEntitiesRequest,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<QueryEntitiesResponse>;
+  // (undocumented)
+  refreshEntity(
+    entityRef: string,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<void>;
+  // (undocumented)
+  removeEntityByUid(
+    uid: string,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<void>;
+  // (undocumented)
+  removeLocationById(
+    id: string,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<void>;
+  // (undocumented)
+  validateEntity(
+    entity: Entity,
+    locationRef: string,
+    options?: CatalogServiceRequestOptions,
+  ): Promise<ValidateEntityResponse>;
+}
+
+// @public
+export const catalogServiceRef: ServiceRef<
+  CatalogService,
+  'plugin',
+  'singleton'
+>;
+
+// @public (undocumented)
+export interface CatalogServiceRequestOptions extends CatalogRequestOptions {
+  // (undocumented)
+  credentials?: BackstageCredentials;
```

Comment:
Should they really be optional?

Reply:
yep could discuss going for the breaking change here instead, might not be too bad tbh. Especially since the catalog client will stay around

Reply:
Updated to only accept and require credentials https://github.com/backstage/backstage/pull/27234/commits/e4e74e720ea85c1d2c3aa3d8044c90244ef8cb10

Bit of an issue with that is that it's hard to support both old and new backend system at the same time with the new service, so we'll need to keep the alpha one around for a while to support that. I think that's an alright tradeoff though